### PR TITLE
[ogre] Add install module Bites

### LINKED ISSUE #12730: [ogre] some headers lost
**Host Environment**
- OS: Win10
- Compiler: VS2017

**To Reproduce**
`./vcpkg install opencv4[ovis]`

**Failure logs** 
`..\buildtrees\opencv4\install-x64-windows-dbg-out.log`:
`..\buildtrees\opencv4\src\4.3.0-795a2a992f.clean\modules\ovis\src\ovis.cpp(7): fatal error C1083: Cannot open include file: 'OgreApplicationContext.h': No such file or directory`

When I check `include\OGRE` folder, and there is no `OgreApplicationContext.h`, while it exists in `%src%\Components\Bites\include`.

### LINKED ISSUE #13632: Issue in Ogre's patch
**Describe the bug**
In file ***fix-dependency.patch***
```
+find_package(freetype CONFIG REQUIRED)
```
The patch searching for package ***freetype***, but forgot to set variables ***FREETYPE_FOUND***, ***FREETYPE_LIBRARIES***,  ***FREETYPE_INCLUDE_DIRS***, which is referenced by ***CMakeLists.txt*** at the line 
```
cmake_dependent_option(OGRE_BUILD_COMPONENT_OVERLAY "Build Overlay component" TRUE "FREETYPE_FOUND" FALSE)
```
This issue causes missing some ogre's components like Overlay or Bites

**Environment**
- OS: Windows
- Compiler: revision MSVC 14.27.29110

**To Reproduce**
1. ./vcpkg install ogre
2. missing ogre's component Overlay

### PR DESCRIPTION
**Pull request by LilyWangL**
**Describe the pull request**

- What does your PR fix? Fixes #12730, Fixes #13632
Installing `opencv4[ovis]` need `ogre-Bites` header files, adding install `ogre-Bites` to fix `opencv4[ovis]` build error.
- Which triplets are supported/not supported? Have you updated the CI baseline?
No
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

**Comment by cenit** [comment]
The same identical patch is required also for opencv3 maybe? Please do not diverge ports reliability

**Comment by LilyWangL** [comment]
> The same identical patch is required also for opencv3 maybe? Please do not diverge ports reliability

I will do it. Thanks for your suggestion.

**Comment by cenit** [comment]
Also, I am almost sure we tested opencv3[ovis] and opencv4[ovis] and they were working. What's changed?
Is the patch not necessary anymore, after 87e0050?

**Comment by LilyWangL** [comment]
> Also, I am almost sure we tested opencv3[ovis] and opencv4[ovis] and they were working. What's changed?
> Is the patch not necessary anymore, after [87e0050](https://github.com/microsoft/vcpkg/commit/87e0050c64fe4c39931b5db48caadffacd205582)?

Yes, it doesn't need add patch on opencv4 and opencv3.

File: ports/ogre/fix-installBites.patch [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
++++ b/CMakeLists.txt
+@@ -332,15 +332,15 @@ cmake_dependent_option(OGRE_BUILD_COMPONENT_TERRAIN "Build Terrain component" TR
+ cmake_dependent_option(OGRE_BUILD_COMPONENT_VOLUME "Build Volume component" TRUE "" FALSE)
+ cmake_dependent_option(OGRE_BUILD_COMPONENT_PROPERTY "Build Property component" TRUE "" FALSE)
+ cmake_dependent_option(OGRE_BUILD_PLUGIN_CG "Build Cg plugin" TRUE "Cg_FOUND;NOT APPLE_IOS;NOT WINDOWS_STORE;NOT WINDOWS_PHONE" FALSE)
+-cmake_dependent_option(OGRE_BUILD_COMPONENT_OVERLAY "Build Overlay component" TRUE "FREETYPE_FOUND" FALSE)
++cmake_dependent_option(OGRE_BUILD_COMPONENT_OVERLAY "Build Overlay component" TRUE "FREETYPE_FOUND" TRUE)
+ 
+-cmake_dependent_option(OGRE_BUILD_COMPONENT_OVERLAY_IMGUI "Include dear imgui in Overlays" TRUE "OGRE_BUILD_COMPONENT_OVERLAY" FALSE)
++cmake_dependent_option(OGRE_BUILD_COMPONENT_OVERLAY_IMGUI "Include dear imgui in Overlays" TRUE "OGRE_BUILD_COMPONENT_OVERLAY" TRUE)
```
Comment (seanyen) [inline ports/ogre/fix-installBites.patch:13]: Why do you need to patch the dependency graph?

**Comment by longhuan2018** [comment]
I think only three changes are needed.
1. In the ogre/portfile.cmake file, add -DFREETYPE_FOUND=ON. Because the freetype cmake file does not set FREETYPE_FOUND
```
vcpkg_configure_cmake(
    SOURCE_PATH ${SOURCE_PATH}
    #PREFER_NINJA
    OPTIONS
        -DOGRE_BUILD_DEPENDENCIES=OFF
        -DOGRE_BUILD_SAMPLES=OFF
        -DOGRE_BUILD_TESTS=OFF
        -DOGRE_BUILD_TOOLS=OFF
        -DOGRE_BUILD_MSVC_MP=ON
        -DOGRE_BUILD_MSVC_ZM=ON
        -DOGRE_INSTALL_DEPENDENCIES=OFF
        -DOGRE_INSTALL_DOCS=OFF
        -DOGRE_INSTALL_PDB=OFF
        -DOGRE_INSTALL_SAMPLES=OFF
        -DOGRE_INSTALL_TOOLS=OFF
        -DOGRE_INSTALL_CMAKE=ON
        -DOGRE_INSTALL_VSPROPS=OFF
        -DOGRE_STATIC=${OGRE_STATIC}
        -DOGRE_CONFIG_THREAD_PROVIDER=std
        -DOGRE_BUILD_RENDERSYSTEM_D3D11=ON
        -DOGRE_BUILD_RENDERSYSTEM_GL=ON
        -DOGRE_BUILD_RENDERSYSTEM_GL3PLUS=ON
        -DOGRE_BUILD_RENDERSYSTEM_GLES=OFF
        -DOGRE_BUILD_RENDERSYSTEM_GLES2=OFF
        **-DFREETYPE_FOUND=ON**
# Optional stuff
        ${FEATURE_OPTIONS}
# vcpkg specific stuff
        -DOGRE_CMAKE_DIR=share/ogre
)
```
2. Modify ${FREETYPE_LIBRARIES} in the fix-findimgui.patch file to freetype, as you did
```
@@ -32,13 +28,13 @@ endif()
 # setup target
 add_library(OgreOverlay ${OGRE_COMP_LIB_TYPE} ${HEADER_FILES} ${SOURCE_FILES})
 set_target_properties(OgreOverlay PROPERTIES VERSION ${OGRE_SOVERSION} SOVERSION ${OGRE_SOVERSION})
-target_link_libraries(OgreOverlay PUBLIC OgreMain PRIVATE ${FREETYPE_LIBRARIES} ZLIB::ZLIB)
+target_link_libraries(OgreOverlay PUBLIC OgreMain PRIVATE **freetype** ZLIB::ZLIB imgui::imgui)
 target_include_directories(OgreOverlay PUBLIC 
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
   $<INSTALL_INTERFACE:include/OGRE/Overlay>
   PRIVATE ${FREETYPE_INCLUDE_DIRS})
```
3. In the Components/Bites/CMakeLists.txt file, add a link to the imgui library. It is recommended to add the modified patch directly to the fix-findimgui.patch file
```
diff --git "a/Components/Bites/CMakeLists.txt" "b/Components/Bites/CMakeLists.txt"
index 5d58a8d74..bd6fe4224 100644
--- "a/Components/Bites/CMakeLists.txt"
+++ "b/Components/Bites/CMakeLists.txt"
@@ -152,6 +152,11 @@ elseif(NOT EMSCRIPTEN)
   message(WARNING "SDL2 not found - no input handling and reduced window creation capabilites")
 endif()
 
+if(OGRE_BUILD_COMPONENT_OVERLAY_IMGUI)
+  find_package(imgui CONFIG REQUIRED)
+  target_link_libraries(OgreBites PRIVATE imgui::imgui)
+endif()
+
 generate_export_header(OgreBites 
     EXPORT_MACRO_NAME _OgreBitesExport
     EXPORT_FILE_NAME ${PROJECT_BINARY_DIR}/include/OgreBitesPrerequisites.h)
```

**Comment by traversaro** [comment]
Hi @LilyWangL, thanks a lot for working on this, as it solves a problem in our downstream CI (see https://github.com/robotology/robotology-superbuild-dependencies-vcpkg/issues/29). There is anything we can do to help to get this merged? Thanks!

**Comment by LilyWangL** [comment]
Closed as duplicate of #14006, this PR also update the version of ogre.

**Pull request closed without merging**